addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -1370,6 +1378,87 @@ MockServer.include({
         messages.length = Math.min(messages.length, limit);
         return this._mockMailMessageMessageFormat(messages.map(message => message.id));
     },
+    /**
+     * Simulates the `/mail/message/add_reaction` route.
+     *
+     * @private
+     * @param {integer} message_id
+     * @param {string} content
+     * @returns {Object}
+     */
+    _mockRouteMailMessageAddReaction(message_id, content) {
+        const partner_id = this.currentPartnerId;
+        this._mockMailMessage_MessageAddReaction(message_id, content);
+        const message_reactions = this._getRecords('mail.message.reaction', [
+            ['message_id', '=', message_id],
+            ['content', '=', content],
+        ]);
+        return {
+            id: message_id,
+            messageReactionGroups: [['insert', {
+                'messageId': message_id,
+                'content': content,
+                'count': message_reactions.length,
+                'partners': [['insert-and-replace', [{ id: partner_id }]]],
+            }]],
+        };
+    },
+    /**
+     * Simulates `_message_add_reaction` on `mail.message`.
+     *
+     * @private
+     * @param {integer} message_id
+     * @param {string} content
+     */
+    _mockMailMessage_MessageAddReaction(message_id, content) {
+        const partner_id = this.currentPartnerId;
+        this._mockCreate('mail.message.reaction', {
+            content,
+            message_id,
+            partner_id,
+        });
+    },
+    /**
+     * Simulates `message_remove_reaction` on `mail.message`.
+     *
+     * @private
+     * @param {integer} message_id
+     * @param {string} content
+     * @returns {Object}
+     */
+    _mockRouteMailMessageRemoveReaction(message_id, content) {
+        const partner_id = this.currentPartnerId;
+        this._mockMailMessage_MessageRemoveReaction(message_id, content);
+        const message_reactions = this._getRecords('mail.message.reaction', [
+            ['message_id', '=', message_id],
+            ['content', '=', content],
+        ]);
```

Comment:
Same remark as other: missing handling of guests

Reply:
Same remark about test env for guests (waiting for @seb-odoo to confirm)